source/tests/system/nisync_session_tests.cpp

```diff
@@ -0,0 +1,118 @@
+#include <gtest/gtest.h>
+
+#include "nisync/nisync_library.h"
+#include "nisync/nisync_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+
+namespace nisync = nisync_grpc;
+
+static const int kSyncDeviceNotFound = -1074118634;
+// Update the value of 'kTestRsrcName' to the name of your NI-Sync device.
+static const char* kTestRsrcName = "PXI1Slot4";
+static const char* kInvalidRsrcName = "InvalidName";
+
+class NiSyncSessionTest : public ::testing::Test {
+ protected:
+  NiSyncSessionTest()
+  {
+    ::grpc::ServerBuilder builder;
+    session_repository_ = std::make_unique<nidevice_grpc::SessionRepository>();
+    nisync_library_ = std::make_unique<nisync::NiSyncLibrary>();
+    nisync_service_ = std::make_unique<nisync::NiSyncService>(nisync_library_.get(), session_repository_.get());
+    builder.RegisterService(nisync_service_.get());
+
+    server_ = builder.BuildAndStart();
+    ResetStubs();
+  }
+
+  virtual ~NiSyncSessionTest() {}
+
+  void ResetStubs()
+  {
+    channel_ = server_->InProcessChannel(::grpc::ChannelArguments());
+    nisync_stub_ = nisync::NiSync::NewStub(channel_);
+  }
+
+  std::unique_ptr<nisync::NiSync::Stub>& GetStub()
+  {
+    return nisync_stub_;
+  }
+
+  ::grpc::Status call_init(const char* resource_name, nisync::InitResponse* response)
+  {
+    ::grpc::ClientContext context;
+    nisync::InitRequest request;
+    request.set_resource_name(resource_name);
+    request.set_reset_device(false);
```

Comment:
After you mark the init function as `'init_method': True,` in the functions.py metadata, it might be useful to exercising setting the session name so add `request.set_session_name(<const char* some_session_name>)`.

Reply:
added test that passes in a session name